python/lsst/ts/atmcssimulator/dataclasses.py

```diff
@@ -0,0 +1,401 @@
+    nasmyth2CalculatedAngle : list[float]
+        Nasmyth2 calculated angle [deg].
+    nasmyth2Encoder1Raw : list[float]
+        Nasmyth2 encoder1 raw value [deg].
+    nasmyth2Encoder2Raw : list[float]
+        Nasmyth2 encoder1 raw value [deg].
+    nasmyth2Encoder3Raw : list[float]
+        Nasmyth2 encoder1 raw value [deg].
+    """
+
+    nasmyth1CalculatedAngle: list[float] = field(default_factory=list)
+    nasmyth1Encoder1Raw: list[float] = field(default_factory=list)
+    nasmyth1Encoder2Raw: list[float] = field(default_factory=list)
+    nasmyth1Encoder3Raw: list[float] = field(default_factory=list)
+    nasmyth2CalculatedAngle: list[float] = field(default_factory=list)
+    nasmyth2Encoder1Raw: list[float] = field(default_factory=list)
+    nasmyth2Encoder2Raw: list[float] = field(default_factory=list)
+    nasmyth2Encoder3Raw: list[float] = field(default_factory=list)
+    trackId: int = 0
+
+    def __post_init__(self) -> None:
+        """Initialize the lists to contain 100 zeroes."""
+        self.nasmyth1CalculatedAngle = [0.0] * 100
+        self.nasmyth1Encoder1Raw = [0.0] * 100
+        self.nasmyth1Encoder2Raw = [0.0] * 100
+        self.nasmyth1Encoder3Raw = [0.0] * 100
+        self.nasmyth2CalculatedAngle = [0.0] * 100
+        self.nasmyth2Encoder1Raw = [0.0] * 100
+        self.nasmyth2Encoder2Raw = [0.0] * 100
+        self.nasmyth2Encoder3Raw = [0.0] * 100
+
+
+@dataclass
+class NasmythM3MountMotorEncoders(Timestamp):
+    """Dataclass holding nasmyth mount motor encoders data.
+
+    Each attributed represents 100 values measured starting at the timestamp.
+
+    Attributes
+    ----------
+    nasmyth1Encoder : list[float]
+        Nasmyth1 encoder value [deg].
+    nasmyth1EncoderRaw : list[float]
+        Nasmyth1 encoder raw value [deg].
+    nasmyth2Encoder : list[float]
+        Nasmyth2 encoder value [deg].
+    nasmyth2EncoderRaw : list[float]
+        Nasmyth2 enccder raw value [deg].
+    m3Encoder : list[float]
+        M3 encoder value [deg].
+    m3EncoderRaw : list[float]
+        M3 encoder raw value [deg].
+    """
+
+    nasmyth1Encoder: list[float] = field(default_factory=list)
+    nasmyth1EncoderRaw: list[float] = field(default_factory=list)
+    nasmyth2Encoder: list[float] = field(default_factory=list)
+    nasmyth2EncoderRaw: list[float] = field(default_factory=list)
+    m3Encoder: list[float] = field(default_factory=list)
+    m3EncoderRaw: list[float] = field(default_factory=list)
+
+    def __post_init__(self) -> None:
+        """Initialize the lists to contain 100 zeroes."""
+        self.nasmyth1Encoder = [0.0] * 100
+        self.nasmyth1EncoderRaw = [0.0] * 100
+        self.nasmyth2Encoder = [0.0] * 100
+        self.nasmyth2EncoderRaw = [0.0] * 100
+        self.m3Encoder = [0.0] * 100
+        self.m3EncoderRaw = [0.0] * 100
+
+
+@dataclass
+class PortInfo:
+    """Dataclass holding port information.
+
+    Attributes
+    ----------
+    index : int
```

Comment:
Please fix the documented types to match the actual type (int | None in this case)